src/Castle.Core/DynamicProxy/ProxyGenerator.cs

```diff
@@ -558,7 +556,7 @@ public object CreateInterfaceProxyWithTarget(Type interfaceToProxy, object targe
 
 			if (target != null)
 			{
-				if (Marshal.IsComObject(target))
+				if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && Marshal.IsComObject(target))
```

Comment:
I am somewhat doubtful that this additional runtime check is truly needed, just because we're adding the `net6.0` TFM.

I understand that improved diagnostics now produce a _compile-time_ warning that wasn't there before, and that's why you're adding this additional _run-time_ check.

However, AFAIK we haven't actually gotten run-time errors in the past due to calling `Marshal.IsComObject` when we should not, so I would tend to simply suppressing that new compiler warning instead of introducing an additional NuGet dependency.

Other opinions?

Reply:
`IsComObject` probably works on any platform as I think it just inspects the .NET type.

I too would prefer a pragma to suppress the warning and leave this without the runtime check.